File: backend/onGo-domain/src/main/kotlin/com/ongo/domain/livestream/LiveStream.kt
package com.ongo.domain.livestream

import java.time.LocalDateTime

data class LiveStream(
    val id: Long? = null,
    val userId: Long,
    val title: String,
    val description: String? = null,
    val platform: String,
    val status: String = "SCHEDULED",
    val scheduledAt: LocalDateTime,
    val startedAt: LocalDateTime? = null,
    val endedAt: LocalDateTime? = null,
    val viewerCount: Int = 0,
    val peakViewers: Int = 0,
    val chatMessages: Int = 0,
    val streamUrl: String? = null,
    val thumbnailUrl: String? = null,
)

File: frontend/src/types/liveStream.ts
export interface LiveStream {
  id: number
  title: string
  description: string
  platform: string
  status: 'SCHEDULED' | 'LIVE' | 'ENDED' | 'CANCELLED'
  scheduledAt: string
  startedAt: string | null
  endedAt: string | null
  viewerCount: number
  peakViewers: number
  chatMessages: number
  streamUrl: string | null
  thumbnailUrl: string | null
}

export interface StreamChat {
  id: number
  streamId: number
  username: string
  message: string
  timestamp: string
  isHighlighted: boolean
  isModerator: boolean
}

export interface LiveStreamSummary {
  totalStreams: number
  liveNow: number
  avgViewers: number
  totalChatMessages: number
  nextScheduled: string | null
}
